src/pykeen/models/nbase.py

```diff
@@ -280,11 +280,7 @@ def __init__(
         self,
         *,
         triples_factory: KGInfo,
-        interaction: Union[
-            str,
-            Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation],
-            Type[Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]],
-        ],
+        interaction: HintOrType[Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]],
```

Comment:
this isn't quite right, doesn't HintOrType allow for none?

Reply:
you want `InstOrType` here since this can't be none

Reply:
well actually `Union[str, InstOrType[...]]` to be the same as the original

Reply:
Alternative: set a default interaction

Reply:
done in https://github.com/pykeen/pykeen/pull/964/commits/42bf9cebfaa5d62b444bc9e077407432cd41de7b